gazebo_ros/src/gazebo_ros_properties.cpp

```diff
@@ -0,0 +1,448 @@
+void GazeboRosPropertiesPrivate::GetJointProperties(
+  gazebo_msgs::srv::GetJointProperties::Request::SharedPtr _req,
+  gazebo_msgs::srv::GetJointProperties::Response::SharedPtr _res)
+{
+  gazebo::physics::JointPtr joint;
+  for (unsigned int i = 0; i < world_->ModelCount(); ++i) {
+    joint = world_->ModelByIndex(i)->GetJoint(_req->joint_name);
+    if (joint) {
+      break;
+    }
+  }
+
+  if (!joint) {
+    _res->success = false;
+    _res->status_message = "GetJointProperties: joint not found";
+  } else {
+    /// @todo: FIXME
+    _res->type = _res->REVOLUTE;
+
+    _res->damping.clear();
+    _res->damping.push_back(joint->GetDamping(0));
+
+    _res->position.clear();
+    _res->position.push_back(joint->Position(0));
+
+    _res->rate.clear();  // use GetVelocity(i)
+    _res->rate.push_back(joint->GetVelocity(0));
+
+    _res->success = true;
+    _res->status_message = "GetJointProperties: got properties";
+  }
+}
+
+void GazeboRosPropertiesPrivate::GetLinkProperties(
+  gazebo_msgs::srv::GetLinkProperties::Request::SharedPtr _req,
+  gazebo_msgs::srv::GetLinkProperties::Response::SharedPtr _res)
+{
+  gazebo::physics::LinkPtr body =
+    boost::dynamic_pointer_cast<gazebo::physics::Link>(world_->EntityByName(_req->link_name));
+  if (!body) {
+    _res->success = false;
+    _res->status_message =
+      "GetLinkProperties: link not found, did you forget to scope the link by model name?";
+  } else {
+    /// @todo: validate
+    _res->gravity_mode = body->GetGravityMode();
+
+    gazebo::physics::InertialPtr inertia = body->GetInertial();
+
+    _res->mass = body->GetInertial()->Mass();
+
+    _res->ixx = inertia->IXX();
+    _res->iyy = inertia->IYY();
+    _res->izz = inertia->IZZ();
+    _res->ixy = inertia->IXY();
+    _res->ixz = inertia->IXZ();
+    _res->iyz = inertia->IYZ();
+
+    ignition::math::Vector3d com = body->GetInertial()->CoG();
+
+    _res->com.position.x = com.X();
+    _res->com.position.y = com.Y();
+    _res->com.position.z = com.Z();
+    _res->com.orientation.x = 0;  // @todo: gazebo do not support rotated inertia yet
+    _res->com.orientation.y = 0;
+    _res->com.orientation.z = 0;
+    _res->com.orientation.w = 1;
+
+    _res->success = true;
+    _res->status_message = "GetLinkProperties: got properties";
+  }
+}
+
+void GazeboRosPropertiesPrivate::GetLightProperties(
+  gazebo_msgs::srv::GetLightProperties::Request::SharedPtr _req,
+  gazebo_msgs::srv::GetLightProperties::Response::SharedPtr _res)
+{
+  gazebo::physics::LightPtr phy_light = world_->LightByName(_req->light_name);
+  if (phy_light == NULL) {
+      _res->success = false;
+      _res->status_message = "getLightProperties: Requested light " + _req->light_name
+       + " not found!";
+  } else {
+    gazebo::msgs::Light light;
+    phy_light->FillMsg(light);
+
+    _res->diffuse.r = light.diffuse().r();
+    _res->diffuse.g = light.diffuse().g();
+    _res->diffuse.b = light.diffuse().b();
+    _res->diffuse.a = light.diffuse().a();
```

Comment:
Consider adding a conversion function from `gazebo::msgs::Color` to `std_msgs::ColorRGBA`, not required though